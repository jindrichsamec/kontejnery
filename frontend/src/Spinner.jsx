import React from 'react'
import Icon from './Icon'
import styled, {keyframes} from 'styled-components'

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`

const Spinner = styled.div`
  display: inline-block;

  & .fa-circle-o-notch {
    animation: ${rotate360} infinite 1s linear;
  }
`

export default () => {
  return (<Spinner>
    <Icon name="circle-o-notch" />
  </Spinner>)
}
