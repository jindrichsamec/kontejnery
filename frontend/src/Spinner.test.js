import Spinner from './Spinner'

it('renders without crashing', () => {
  const wrapper = shallow(<Spinner />);
  expect(wrapper).toMatchSnapshot();
});
