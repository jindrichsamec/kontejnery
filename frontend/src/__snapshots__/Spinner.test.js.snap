// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<styled.div>
  <Icon
    name="circle-o-notch"
  />
</styled.div>
`;
